osu.Game/Online/Rooms/BeatmapAvailability.cs

```diff
@@ -19,17 +19,17 @@ public class BeatmapAvailability : IEquatable<BeatmapAvailability>
         /// <summary>
         /// The beatmap's downloading progress, null when not in <see cref="DownloadState.Downloading"/> state.
         /// </summary>
-        public readonly double? DownloadProgress;
+        public readonly float? DownloadProgress;
 
         [JsonConstructor]
-        private BeatmapAvailability(DownloadState state, double? downloadProgress = null)
+        private BeatmapAvailability(DownloadState state, float? downloadProgress = null)
```

Comment:
Should this not have been merged and part of the server-side deploy? If we were using anything other than json serialisation this is an API breaking change.

Reply:
Right, was concerned about that. this is using JSON serialization, but for safety, I'll move this to a different PR and let server depend on it.

Reply:
It's not worth it at this point, fine to stay here. Just pointing it out.